### PR TITLE
Replace documentation mentions of IRC with Discord

### PR DESCRIPTION
**Pull request by AaronBallman**
This does not touch code owners or credits files that list IRC handles, that can be done separately if we want to make that change.

See https://discourse.llvm.org/t/rfc-remove-irc-as-a-recommended-communication-channel/82808/3 for the RFC.

### CODE REVIEW

**Comment by AaronBallman** [comment]
The developer policy got nailed by line ending changes; I'll try to redo that so it's more obvious what's changed.

File: llvm/docs/GettingInvolved.rst [hunk 1]
```diff
@@ -377,7 +377,6 @@ Guidance for office hours hosts
 
     * the `#office-hours Discord channel
```
Comment (erichkeane) [inline llvm/docs/GettingInvolved.rst:378]: Weird to have a list of 1.

File: llvm/docs/GettingInvolved.rst [hunk 2]
```diff
@@ -388,27 +387,18 @@ Guidance for office hours hosts
...
-build statuses, but it is a noisy channel due to the automated messages. The
-channel is not actively moderated.
-
-In addition to the traditional IRC there is a
-`Discord <https://discord.com/channels/636084430946959380/636725486533345280>`_
-chat server available. To sign up, please use this
-`invitation link <https://discord.com/invite/xS7Z362>`_.
+bot will update the channel with a link to a build bot when a build goes from
+passing to failing and again when the build goes from failing back to passing.
+It is a good channel for actively monitoring build statuses.
```
Comment (erichkeane) [inline llvm/docs/GettingInvolved.rst:401]: Perhaps something like, "It is a great way to actively monitor the status of the build"? 

Else this sentence reads awkward.

**Review by erichkeane — COMMENTED** [review]
Yikes on the Dev policy, but reviewed the rest.

File: llvm/docs/DeveloperPolicy.rst [hunk 3]
```diff
@@ -201,7 +201,7 @@ within an area of a project:
     * aid release managers with backporting and other release-related
       activities,
     * be a point of contact for contributors who need help (answering questions
-      on Discord/Discourse/IRC or holding office hours).
+      on Discord/Discourse/Discord or holding office hours).
```
Comment (erichkeane) [inline llvm/docs/DeveloperPolicy.rst:204]: Discord or more discord?

File: llvm/docs/GettingInvolved.rst (hunk 1 above)
Comment (AaronBallman) [reply llvm/docs/GettingInvolved.rst:378]: fair point, I'll fix that

**Review by jrtc27 — APPROVED** [review]
LGTM subject to the existing review comments

**Review by erichkeane — APPROVED** [review]

File: llvm/docs/GettingInvolved.rst [hunk 4]
```diff
@@ -388,27 +384,18 @@ Guidance for office hours hosts
   from the list above.
 
 
-.. _IRC:
-
-IRC
----
+Discord
```
Comment (jcranmer-intel) [inline llvm/docs/GettingInvolved.rst:387]: You've removed the thing that lets you link to this section, if I'm recalling my RST correctly.

**Review by jcranmer-intel — COMMENTED** [review]
One thing I have noticed is that this does seem to more or less by accident remove the IRC channel from the list of communication channels to which the code of conduct and other policies applied.

I am on board with generally replacing use of IRC with Discord, and preferring discord especially in our documentation, but I wonder if it might make sense to have some residual mention somewhere that #llvm on the OFTC network as being project-affiliated that could cover that case (even if we emphasize that it's not particularly active). I don't have any good wording suggestions, I'm afraid.

File: llvm/docs/CodeOfConduct.rst [hunk 5]
```diff
@@ -24,12 +24,12 @@ the spirit in which it's intended - a guide to make it easier to communicate
 and participate in the community.
 
 This code of conduct applies to all spaces managed by the LLVM project or The
-LLVM Foundation. This includes IRC channels, mailing lists, bug trackers, LLVM
-events such as the developer meetings and socials, and any other forums created
-by the project that the community uses for communication. It applies to all of
-your communication and conduct in these spaces, including emails, chats, things
-you say, slides, videos, posters, signs, or even t-shirts you display in these
-spaces. 
+LLVM Foundation. This includes Discord channels, mailing lists, bug trackers,
```
Comment (kbeyls) [inline llvm/docs/CodeOfConduct.rst:27]: Regarding the comment on not explicitly mentioning anymore that the code of conduct continues to apply to the IRC channel too: Probably the easiest way to address that is to keep IRC in the enumerated list here.
IRC is a "legacy" channel, but it still exists, so the CoC continues to apply there.

**Review by kbeyls — COMMENTED** [review]
Overall looks great, thank you :)
I've left one nit-picky comment on keeping IRC mentioned in the code of conduct page, as even though it's a "legacy" communication channel, it continues to exist, so the code-of-conduct continues to apply there.

**Review by kbeyls — APPROVED** [review]

**Comment by nickdesaulniers** [comment]
LGTM

**Pull request merged**